src/core/evaluator.js

```diff
@@ -756,9 +766,13 @@ class PartialEvaluator {
       localColorSpaceCache,
     })
       .then(imageObj => {
-        imgData = imageObj.createImageData(/* forceRGBA = */ false);
+        imgData = imageObj.createImageData(
+          /* forceRGBA = */ false,
+          /* isOffscreenCanvasSupported = */ this.options
+            .isOffscreenCanvasSupported
+        );
 
-        if (cacheKey && imageRef && cacheGlobally) {
+        if (cacheKey && imageRef && cacheGlobally && imgData.data) {
           this.globalImageCache.addByteSize(imageRef, imgData.data.length);
         }
```

Comment:
This would allow us to cache an *unbounded* number of images globally, which could affect memory usage really badly, if we simply ignore the `bitmap` case here.

It thus seems to me that you need to determine the correct length here, [similar to this code](https://github.com/mozilla/pdf.js/blob/546902df63b4fcfad419bc8109d128040d7b27ab/src/display/api.js#L2774-L2781):
```suggestion
        if (cacheKey && imageRef && cacheGlobally) {
          let length = 0;
          if (imgData.bitmap) {
            length = imgData.width * imgData.height * 4;
          } else {
            length = imgData.data.length,
          }
          this.globalImageCache.addByteSize(imageRef, length);
        }
```